src/main/java/com/iexec/worker/pubsub/SubscriptionService.java

```diff
@@ -191,7 +191,7 @@ private void handleTaskNotification(TaskNotification notif) {
                     break;
 
                 case PLEASE_REVEAL:
-                    taskExecutorService.reveal(chainTaskId);
+                    taskExecutorService.reveal(chainTaskId, notif.getBlockNumber());
```

Comment:
notif.getConsensusBlock()

Reply:
the block number is generic depending on the taskNotificationType.

Reply:
Yes you'r right I thought that few minutes later!